problem_builder/swipe.py

```diff
@@ -0,0 +1,171 @@
+# WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU Affero
+# General Public License for more details.
+#
+# You should have received a copy of the GNU Affero General Public License
+# along with this program in a file in the toplevel directory called
+# "AGPLv3".  If not, see <http://www.gnu.org/licenses/>.
+#
+
+# Imports ###########################################################
+import logging
+
+from xblock.fields import Scope, String
+from xblock.validation import ValidationMessage
+from xblockutils.resources import ResourceLoader
+
+from .mixins import StudentViewUserStateMixin
+from .questionnaire import QuestionnaireAbstractBlock
+from .sub_api import SubmittingXBlockMixin
+
+
+# Globals ###########################################################
+log = logging.getLogger(__name__)
+loader = ResourceLoader(__name__)
+
+
+# Make '_' a no-op so we can scrape strings
+def _(text):
+    return text
+
+
+# Classes ###########################################################
+class SwipeBlock(SubmittingXBlockMixin, StudentViewUserStateMixin, QuestionnaireAbstractBlock):
+    """
+    An XBlock used to ask binary-choice questions with a swiping interface
+    """
+    CATEGORY = 'pb-swipe'
+    STUDIO_LABEL = _(u"Swipeable Binary Choice Question")
+    USER_STATE_FIELDS = ['num_attempts', 'student_choice']
+
+    message = String(
+        display_name=_("Message"),
+        help=_(
+            "General feedback provided when submitting. "
+            "(This is not shown if there is a more specific feedback tip for the choice selected by the learner.)"
+        ),
+        scope=Scope.content,
+        default=""
+    )
+
+    student_choice = String(
+        # {Last input submitted by the student
+        default="",
+        scope=Scope.user_state,
+    )
+
+    correct_choice = String(
+        display_name=_("Correct Choice"),
+        help=_("Specify the value that students may select for this question to be considered correct."),
+        scope=Scope.content,
+        values_provider=QuestionnaireAbstractBlock.choice_values_provider,
+    )
+    img_url = String(
+        display_name=_("Image"),
+        help=_(
+            "Specify the URL of an image associated with this question."
+        ),
+        scope=Scope.content,
+        default=""
+    )
+    editable_fields = QuestionnaireAbstractBlock.editable_fields + ('message', 'correct_choice', 'img_url',)
+
+    def calculate_results(self, submission):
+        correct = self.correct_choice == submission
+        return {
+            'submission': submission,
+            'message': self.message_formatted,
+            'status': 'correct' if correct else 'incorrect',
+            'weight': self.weight,
+            'score': 1 if correct else 0,
+        }
+
+    def get_results(self, previous_result):
+        return self.calculate_results(self, previous_result['submission'])
```

Comment:
Submitting the answer now works, but I'm getting "TypeError: calculate_results() takes exactly 2 arguments (3 given)" if I reload the page in the LMS after an answer has been submitted.

It should be `calculate_results(previous_result['submission'])` instead of `calculate_results(self, ...)`.